PRIMER ENTREGA

En esta primer entrega presentaré el set de datos elegido para el proyecto final, que es un registro de accidentes viales de la Ciudad de Buenos Aires (desde ahora CABA), descargado de la pagina oficial del Gobierno de la Ciudad.

En este primer analisis veremos los detalles de este dataset, como por ejemplo el periodo de tiempo que abarca, los datos geograficos que nos dan, con que detalles del accidente contamos, etc. 

La motivación de esta elección es el estudio de casos reales y cercanos a mi en cierto punto (ya que vivo cerca de la ciudad de buenos aires y trabajo ahí) que me permitirá plantear preguntas interesantes y sacar conclusiones usando mi conocimiento de la Ciudad y sus conductores/transeuntes.

La audiencia que se vería beneficiada con este analisis es el ciudadano de CABA, que está expuesto a los infractores y el peligro que estos suponen, ya que, como opinión personal, los conductores y peatones de CABA tienen una tendencia general al desinteres hacia el projimo en las circunstancias de la vialidad.

Las preguntas que haré a lo largo de la entrega se basaran en el estudio del set de datos, pero como primer objetivo a tener en cuenta, quiero reducir la cantidad de siniestros en la Ciudad de Buenos Aires, con medidas que utilicen la información presente en los datos del dataset, para esto asumo que me contrata el Gobierno de la Ciudad. 

IMPORTO LAS LIBRERIAS A UTILIZAR.

LUEGO IMPORTO Y CARGO EL SET DE DATOS EN UN DATAFRAME LLAMADO "sin_df".

In [1]:
import pandas as pd

In [5]:
siniestros = 'victimas-siniestros.csv'
sin_df = pd.read_csv(siniestros)
sin_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33234 entries, 0 to 33233
Data columns (total 29 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   causa                         33234 non-null  object 
 1   rol                           32899 non-null  object 
 2   tipo                          32134 non-null  object 
 3   sexo                          33037 non-null  object 
 4   edad                          32675 non-null  float64
 5   mes                           33197 non-null  float64
 6   periodo                       33234 non-null  int64  
 7   fecha                         33173 non-null  object 
 8   hora                          33217 non-null  object 
 9   lugar_hecho                   29518 non-null  object 
 10  direccion_normalizada         32469 non-null  object 
 11  tipo_calle                    33234 non-null  object 
 12  direccion_normalizada_arcgis  32462 non-null  object 
 13  c

C:\Users\Seguridad\AppData\Local\Temp\ipykernel_17220\2868349731.py:2: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.
  sin_df = pd.read_csv(siniestros)


Tengo 29 columnas en el dataset, en esta entrega no haré la limpieza de las columnas, pero si una exploración para entender cuales son las mas importantes y cuales son redundantes.

In [25]:
datos_sin = sin_df[['causa','rol','tipo','cantidad_victimas','participantes_victimas','participantes_acusados','tipo_calle','tipo_colision1']]
datos_pers = sin_df[['rol','sexo','edad']]
datos_fecha_geo = sin_df[['mes','periodo','fecha','hora','lugar_hecho','direccion_normalizada','direccion_normalizada_arcgis','calle1','altura','calle2','codigo_calle','codigo_cruce','geocodificacion','semestre','x','y','geom','comuna','geom_3857']]

El dataset me ofrece datos del siniestro, de la persona involucrada y de la fecha y ubicación del hecho. Divido estos datos en 3 dataframes para analizar estos y encontrar redundancias.

In [15]:
datos_sin.head(10)

causa        rol       tipo  cantidad_victimas participantes_victimas  \
0  homicidio  conductor       moto                  1                   moto   
1  homicidio        NaN        NaN                  1                    NaN   
2  homicidio     peaton     peaton                  1                 peaton   
3  homicidio        NaN        NaN                  1                    NaN   
4  homicidio   pasajero     camion                  1                 camion   
5  homicidio  conductor       moto                  1                   moto   
6  homicidio  conductor       moto                  1                   moto   
7   lesiones   pasajero  automovil                  1              automovil   
8  homicidio  conductor       moto                  1                   moto   
9  homicidio        NaN       moto                  1                   moto   

  participantes_acusados tipo_calle               tipo_colision1  
0              automovil      calle      motovehiculo - vehiculo  
1                    NaN    avenida                          NaN  
2   "transporte publico"    avenida            peaton - vehiculo  
3                    NaN      calle                          NaN  
4                 camion    avenida          vehiculo - vehiculo  
5              automovil      calle      motovehiculo - vehiculo  
6              automovil    avenida      motovehiculo - vehiculo  
7              automovil      calle          vehiculo - vehiculo  
8                   moto    avenida  motovehiculo - motovehiculo  
9              automovil    avenida      motovehiculo - vehiculo

In [16]:
datos_sin['causa'].value_counts()

lesiones     32795
homicidio      439
Name: causa, dtype: int64

En el dataframe con datos del siniestro veo, en principio, que todas las columnas son importantes y tienen información sensible no redundante. 
También puedo notar que los registros que tienen algúna columna con dato nulo, es muy probable que tengan otras columnas con nulos también (por ejemplo los registros del indice 1 y 3).

Por alguna razón que todavía no interpreto, los primeros registros tienen una gran cantidad de valores "homicidio" en la columna "causa", a pesar de que este valor aparezca solo 439 veces (mientras que el otro valor posible, "lesiones", aparece 32795 veces). Es posible que lo hayan cargado de forma no cronologica, esto lo confirmare estudiando los registros que involucran la fecha y localización.

In [17]:
datos_pers.head(10)

rol       sexo  edad
0  conductor        NaN   NaN
1        NaN        NaN   NaN
2     peaton   femenino   NaN
3        NaN        NaN   NaN
4   pasajero  masculino   NaN
5  conductor        NaN   NaN
6  conductor  masculino  18.0
7   pasajero  masculino  26.0
8  conductor  masculino  24.0
9        NaN  masculino   NaN

In [20]:
datos_pers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33234 entries, 0 to 33233
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   rol     32899 non-null  object 
 1   sexo    33037 non-null  object 
 2   edad    32675 non-null  float64
dtypes: float64(1), object(2)
memory usage: 779.0+ KB


En este caso veo muchos datos nulos a primera vista, pero usando la función info puedo notar que es un problema en el orden de la carga de los registros, ya que las 3 columnas tienen la mayoria de sus datos no nulos, como pude ver arriba. Teniendo en cuenta lo notado previamente sobre la gran carga de registros en los que hubo fallecidos al principio del set de datos, esta particularidad de la columna edad con datos nulos no parece ser una coincidencia, sino que en principio habría una carga con una mayor cantidad de datos nulos en estos casos en los que el participante del siniestro fallecio (y estos parecen estar en los primeros registros del set de datos) 

In [23]:
datos_pers.sample(5)

rol       sexo  edad
21136  conductor   femenino  50.0
10477   pasajero   femenino  41.0
21583  conductor  masculino  18.0
26294     peaton   femenino  33.0
6826   conductor  masculino  37.0

Con el metodo sample confirmo la sospecha expuesta arriba (ya que no obtengo nulos).

In [30]:
datos_fecha_geo.sample(5)

mes  periodo       fecha      hora  \
29517  10.0     2016  10/19/2016  16:50:00   
6107    6.0     2017   6/29/2017  14:20:00   
1076   12.0     2015    12-11-15  19:50:00   
15493  12.0     2017  12/15/2017  14:45:00   
14900  11.0     2017  11/14/2017  17:15:00   

                                     lugar_hecho  \
29517     guatemala & avda raul scalabrini ortiz   
6107                    5452 avda juan b alberdi   
1076                         av. san martin 1460   
15493                   salguero y guardia vieja   
14900  acceso colectora gral paz y autp gral paz   

                        direccion_normalizada  \
29517  guatemala y scalabrini ortiz, raul av.   
6107          alberdi, juan bautista av. 5452   
1076                      san martin av. 1460   
15493      salguero, jeronimo y guardia vieja   
14900                                     NaN   

                 direccion_normalizada_arcgis                      calle1  \
29517  guatemala & scalabrini ortiz, raul av.                   guatemala   
6107          5452 alberdi, juan bautista av.  alberdi, juan bautista av.   
1076                      1460 san martin av.              san martin av.   
15493      salguero, jeronimo & guardia vieja          salguero, jeronimo   
14900                                     NaN                         NaN   

       altura                      calle2  codigo_calle  codigo_cruce  \
29517     NaN  scalabrini ortiz, raul av.        7093.0       20133.0   
6107   5452.0                         NaN        1033.0           NaN   
1076   1460.0                         NaN       20045.0           0.0   
15493     NaN               guardia vieja           NaN           NaN   
14900     NaN                         NaN           NaN           NaN   

                               geocodificacion  semestre          x  \
29517  POINT(103835.665024559 104509.35389737)         2 -58.421491   
6107        point(96688.2500552 97823.1969389)         1 -58.499424   
1076         POINT(101214.511103 102498.27935)         2 -58.450059   
15493  point(104047.25008821 103153.526689802)         2 -58.419178   
14900                                      NaN         2        NaN   

               y                                               geom  comuna  \
29517 -34.588614  0101000020E6100000DF2F2E68F3354DC0EED6B8B3574B...    14.0   
6107  -34.648887  0101000020E6100000EB9E7222ED3F4DC00EB06EBA0E53...     9.0   
1076  -34.606749  0101000020E61000002D3101859B394DC079C46CF2A94D...     6.0   
15493 -34.600835  0101000020E6100000EE6D35A0A7354DC06747822AE84C...     5.0   
14900        NaN                                                NaN     NaN   

                                               geom_3857  
29517  0101000020110F0000DCC615A606CF58C161D75981A957...  
6107   0101000020110F0000D62DCA877FD758C1D0CF9F089667...  
1076   0101000020110F00009C7582B021D258C1E36CE5BC735C...  
15493  0101000020110F0000D0D8AE49C6CE58C19D63BFD6E35A...  
14900                                                NaN

Acá tengo varias cosas para resaltar, primero, la columna fecha tiene dos posibles formatos para sus valores (DD/MM/AAAA y MM-DD-AA), debo normalizar esto para poder extraer el dia y usarlo como columna. Por otro lado, la geolocalización tiene muchas columnas redundantes (con información repetida en distintos formatos), por lo que hare una limpieza y eliminare las columnas que no aporten información. 

A CONTINUACIÓN VOY A EXPLORAR EL SET DE DATOS PARA TERMINAR DE ENTENDER COMO FUE LA CARGA DE DATOS Y LAS RELACIONES ENTRE LOS DATOS NULOS.

En primer lugar corroboro la relación entre el valor 'homicidio' de la columna 'causa' con los datos nulos e intento entender por que pareciera que estan distribuidos en los primeros registros del set de datos.

In [37]:
sin_df.loc[sin_df.causa == 'homicidio'].describe()

edad         mes      periodo        altura  codigo_calle  \
count  387.000000  439.000000   439.000000     73.000000    406.000000   
mean    39.418605    6.808656  2015.970387   2949.315068  15682.714286   
std     19.601086    3.563490     0.817356   2562.321554  20794.019197   
min      1.000000    1.000000  2015.000000     30.000000   1006.000000   
25%     24.000000    4.000000  2015.000000   1211.000000   4056.000000   
50%     34.000000    7.000000  2016.000000   2186.000000  12107.000000   
75%     52.000000   10.000000  2017.000000   4386.000000  17118.500000   
max    100.000000   12.000000  2017.000000  15200.000000  99657.000000   

       codigo_cruce    semestre           x           y  cantidad_victimas  \
count    316.000000  439.000000  435.000000  435.000000         439.000000   
mean   13387.132911    1.519362  -58.441096  -34.621098           1.086560   
std    14447.084076    0.500195    0.046894    0.036720           0.326566   
min     1014.000000    1.000000  -58.528953  -34.703799           1.000000   
25%     4077.250000    1.000000  -58.475866  -34.648065           1.000000   
50%    12107.000000    2.000000  -58.442734  -34.624472           1.000000   
75%    19048.000000    2.000000  -58.399077  -34.597962           1.000000   
max    99657.000000    2.000000  -58.358625  -34.534649           3.000000   

           comuna  
count  435.000000  
mean     7.321839  
std      4.373923  
min      1.000000  
25%      4.000000  
50%      8.000000  
75%     11.000000  
max     15.000000

In [ ]:
sin_df.loc[sin_df.causa == 'homicidio'].